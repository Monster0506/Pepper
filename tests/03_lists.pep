%% Test: List Operations

%% List creation and basic operations
LET numbers: list = [1, 2, 3, 4, 5]
SHOW(numbers)

%% Append
REAS numbers = numbers 6 [a]
SHOW(numbers)

%% Remove
REAS numbers = numbers 3 [r]
SHOW(numbers)

%% Insert
REAS numbers = numbers 10 [n] 2
SHOW(numbers)

%% RePlace first occurrence
REAS numbers = numbers 4 [p] 40
SHOW(numbers)

%% RePlace all occurrences
REAS numbers = numbers 5 [P] 50
SHOW(numbers)

%% List length
LET length: int = numbers [l]
SHOW(length)

%% List indexing
LET item: int = numbers [i] 2
SHOW(item)

%% List from string
LET text: string = "Hello"
LET chars: list = text :> list
SHOW(chars)

%% Multiple operations
REAS numbers = numbers 100 [a]
REAS numbers = numbers 100 [P] 200
SHOW(numbers)
