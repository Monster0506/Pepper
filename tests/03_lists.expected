[1, 2, 3, 4, 5]
[1, 2, 3, 4, 5, 6]
[1, 2, 4, 5, 6]
[1, 10, 2, 4, 5, 6]
[1, 10, 2, 40, 5, 6]
[1, 10, 2, 40, 50, 6]
6
10
['H', 'e', 'l', 'l', 'o']
[1, 10, 2, 40, 50, 6, 200]
5
0
[1, 10, 2, 40, 50, 6, 200]
50
1